code/datums/traits/negative.dm

```diff
@@ -100,94 +100,339 @@
 	if(DT_PROB(0.05, delta_time))
 		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "depression", /datum/mood_event/depression)
 
+
 /datum/quirk/family_heirloom
 	name = "Family Heirloom"
 	desc = "You are the current owner of an heirloom, passed down for generations. You have to keep it safe!"
-	value = -1
+	value = -2
 	mood_quirk = TRUE
 	var/obj/item/heirloom
 	var/where
 
 /datum/quirk/family_heirloom/on_spawn()
 	var/mob/living/carbon/human/H = quirk_holder
 	var/obj/item/heirloom_type
+	var/list/heirloom_table = list()
+	var/civ = FALSE
+	var/med = FALSE
+	var/sec = FALSE
+	var/sci = FALSE
+	var/engi = FALSE
+	var/supp = FALSE
+	var/comm = FALSE
+
+	//------------------------------//
+	// Adding items from this point //
+	//------------------------------//
+
+	// 1. Species specific table
+	// '|| prob(1)' means you can get other race's heirloom at low chance.
+	//Human
+	if(is_species(H, /datum/species/human) || prob(1))
+		heirloom_table.Add(/obj/item/clothing/head/kitty) //you had terrible parents.
+	//Felinid
+	if(is_species(H, /datum/species/human/felinid) || prob(1))
+		heirloom_table.Add(/obj/item/clothing/head/kitty)
+	//Lizard
+	if(is_species(H, /datum/species/lizard) || prob(1))
+		heirloom_table.Add(/obj/item/toy/plush/lizardplushie,
+						   /obj/item/clothing/under/costume/gladiator)
+	//Oozeling
+	if(is_species(H, /datum/species/oozeling) || prob(1))
+		heirloom_table.Add(/obj/item/toy/plush/slimeplushie)
+	//Moth
+	if(is_species(H, /datum/species/moth) || prob(1))
+		heirloom_table.Add(/obj/item/flashlight/lantern/heirloom_moth,
+						   /obj/item/toy/plush/moth)
+	//Apid
+	if(is_species(H, /datum/species/apid) || prob(1))
+		heirloom_table.Add(/obj/item/toy/plush/beeplushie)
+	//Plamaman
+	if(is_species(H, /datum/species/plasmaman) || prob(1))
+		heirloom_table.Add(/obj/item/coin/plasma)
+	//Ethereal
+	if(is_species(H, /datum/species/ethereal) || prob(1))
+		heirloom_table.Add(/obj/item/coin/plasma) //I am not sure what to give them
+	//IPC
+	if(is_species(H, /datum/species/ipc) || prob(1))
+		heirloom_table.Add(/obj/item/disk/data)
+	//Grod
+	//*TBD
+	//if(is_species(H, /datum/species/grod) || prob(1))
+	//	heirloom_table.Add(TBD)
+
+	//retired beecode speices, but let's give them some love
+	//Fly
+	if(is_species(H, /datum/species/fly) || prob(1))
+		heirloom_table.Add(/obj/item/reagent_containers/food/drinks/bottle/virusfood)
+	//Squid
+	//if(is_species(H, /datum/species/squid) || prob(1))
+	//	heirloom_table.Add(/obj/item/toy/eldrich_book)
+
+
+	// 2. Job specific table
+	switch(quirk_holder.mind.assigned_role)
+		//Service jobs
+		if("Assistant")
+			civ = TRUE
+			heirloom_table.Add(/obj/item/storage/toolbox/mechanical/old/heirloom,
+						   	   /obj/item/clothing/gloves/cut/heirloom,
+						   	   /obj/item/multitool)
+			if(prob(1) && prob(50))
```

Comment:
Prob supports floating point values. (prob(0.5) = prob(1) && prob(50))
![image](https://user-images.githubusercontent.com/26465327/143591171-f531740c-22cd-4e2a-bd01-c8925de168d9.png)

Reply:
oh, I thought they don't support float since I've seen such things in the code. thanks!